lib/client/credential/idemix/credential.go

```diff
@@ -108,14 +120,31 @@ func (cred *Credential) Load() error {
 		log.Debugf("No credential found at %s: %s", cred.signerConfigFile, err.Error())
 		return err
 	}
-	val := SignerConfig{}
-	err = json.Unmarshal(signerConfigBytes, &val)
-	if err != nil {
-		return errors.Wrapf(err, fmt.Sprintf("Failed to unmarshal SignerConfig bytes from %s", cred.signerConfigFile))
+
+	// Load the MSP signer config
+	var val SignerConfig
+	mspSignerConfig := &m.IdemixMSPSignerConfig{}
+	err = proto.Unmarshal(signerConfigBytes, mspSignerConfig)
+	if err == nil {
+		val = SignerConfig{
+			Cred:                            mspSignerConfig.Cred,
+			Sk:                              mspSignerConfig.Sk,
+			OrganizationalUnitIdentifier:    mspSignerConfig.OrganizationalUnitIdentifier,
+			Role:                            int(mspSignerConfig.Role),
+			EnrollmentID:                    mspSignerConfig.EnrollmentId,
+			CredentialRevocationInformation: mspSignerConfig.CredentialRevocationInformation,
+		}
 	}
-	if val.CurveID == "" {
-		val.CurveID = idemix4.DefaultIdemixCurve
```

Comment:
Don't we still have to handle the blank val.CurveID?

Reply:
that field is not needed anymore